rewrite-java-21/src/main/java/org/openrewrite/java/isolated/ReloadableJava21ParserVisitor.java

```diff
@@ -348,13 +348,15 @@ public J visitCase(CaseTree node, Space fmt) {
                 Markers.EMPTY,
                 type,
                 null,
+                null,
                 JContainer.build(
-                        node.getExpressions().isEmpty() ? EMPTY : sourceBefore("case"),
-                        node.getExpressions().isEmpty() ?
+                        node.getLabels().isEmpty() ? EMPTY : sourceBefore("case"),
+                        node.getLabels().isEmpty() || node.getLabels().getFirst() instanceof DefaultCaseLabelTree ?
                                 List.of(JRightPadded.build(new J.Identifier(randomId(), Space.EMPTY, Markers.EMPTY, emptyList(), skip("default"), null, null))) :
-                                convertAll(node.getExpressions(), commaDelim, t -> EMPTY),
+                                convertAll(node.getLabels(), commaDelim, ignored -> sourceBefore("when", '-')),
```

Comment:
I know I suggested this, but thinking about it I think this is cleaner and safer (untested):

```suggestion
                                convertAll(node.getLabels(), commaDelim, ignored -> node.getGuard() != null ? sourceBefore("when") : EMPTY),
```